Realm/RLMProperty.mm

```diff
@@ -343,6 +343,9 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
     }
     else if ([_objcRawType isEqualToString:@"@\"NSNumber\""]) {
         const char *numberType = [propertyValue objCType];
+        if (!numberType) {
+            @throw RLMException(@"Can't persist NSNumber directly: use a Swift-native number type.");
```

Comment:
Ooohhh, I misunderstood the @mrackwitz comment. I had thought it was a suggested alternative implementation, but I now understand that it was an addition to the exception message. Sorry for the confusion!